omics/uriparse/uri_parse.py

```diff
@@ -30,8 +30,15 @@ def __init__(self, omics_uri):
         if url_parts.scheme != self.SCHEME:
             raise ValueError(f"Invalid URI scheme, expected {self.SCHEME}")
 
-        if self.NETLOC not in url_parts.netloc:
-            raise ValueError(f"Invalid URI netloc, expected {self.NETLOC}")
+        netloc_parts = url_parts.netloc.split(".")
+        if len(netloc_parts) != 5 or netloc_parts[1] != self.NETLOC:
```

Comment:
Have you considered using Regex Capture Groups to validate and parse the URL?  Seems like that could handle edge cases like unusual characters in the Account ID position and be more sophisticated than `split` and `strip`.

Not a requirement to merge - just worth discussing.

Reply:
Yes, I was also worried about the open-ended parsing we're doing here and would prefer a regex. I'll update with those changes.